### DISCUSSION TITLE
Authentication by scanning a qr code

### DISCUSSION THREAD

**Discussion by Romain-Fromantin** [ory/kratos#2650] [Ideas]
Hello, 

(I have not found any documentation on this subject) 

it would be nice if users could authenticate themselves by scanning a qr code with their phones (from the apps)
(like discord)

<a href="https://ibb.co/LrZbqp9"><img src="https://i.ibb.co/vhjgrzL/scanning-qr-code.gif" alt="scanning-qr-code" border="0"></a>
_click on the image to see the gif_

**Comment by vinckr** [comment]
Hello @romain63110,

this should be possible using WebAuthn - although I have not tried it. 
We most likely just need to enable some additional settings in WebAuthn for this to work out of the box - would you be open to [file this as a feature request](https://github.com/ory/kratos/issues/new?assignees=&labels=feat&template=FEATURE-REQUEST.yml) so we can keep track of it? That would help a lot!
Thanks :)

**Reply by Romain-Fromantin** [reply]
the qr code in webauthn allows you to add an device (such as a security key)?

I was thinking of an API that allows you to authenticate (example: from a computer) by scanning a qr code from a device that is already authenticated (example:phone(application)).
This avoids typing the username.
